src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3110,9 +3110,9 @@ internal struct Buffer<TElement>
 
         internal Buffer(IEnumerable<TElement> source, bool queryInterfaces = true)
         {
-            TElement[] items = null;
-            int count = 0;
-
+            items = null;
+            count = 0;
```

Comment:
With this change, all of the accesses in this method are now field accesses rather than local accesses.  You've verified that doesn't harm the generated assembly?